packages/iframe-execution-environment-service/src/IframeExecutionEnvironmentService.ts

```diff
@@ -0,0 +1,337 @@
+import { Duplex } from 'stream';
+import { nanoid } from 'nanoid';
+import pump from 'pump';
+import { ObservableStore } from '@metamask/obs-store';
+import ObjectMultiplex from '@metamask/object-multiplex';
+import { WindowPostMessageStream } from '@metamask/post-message-stream';
+import { PLUGIN_STREAM_NAMES } from '@mm-snap/workers';
+import { createStreamMiddleware } from 'json-rpc-middleware-stream';
+import { PluginData } from '@mm-snap/types';
+import {
+  JsonRpcEngine,
+  JsonRpcRequest,
+  PendingJsonRpcResponse,
+} from 'json-rpc-engine';
+import { ExecutionEnvironmentService } from '@mm-snap/controllers';
+
+export type SetupPluginProvider = (pluginName: string, stream: Duplex) => void;
+
+interface JobControllerArgs {
+  setupPluginProvider: SetupPluginProvider;
+  iframeUrl: URL;
+}
+
+interface JobStreams {
+  command: Duplex;
+  rpc: Duplex | null;
+  _connection: WindowPostMessageStream;
+}
+
+// The plugin is the callee
+export type PluginRpcHook = (
+  origin: string,
+  request: Record<string, unknown>,
+) => Promise<unknown>;
+
+interface EnvMetadata {
+  id: string;
+  streams: JobStreams;
+  rpcEngine: JsonRpcEngine;
+}
+
+export class IframeExecutionEnvironmentService
+  implements ExecutionEnvironmentService
+{
+  public store: ObservableStore<{ jobs: Record<string, EnvMetadata> }>;
```

Comment:
I know this is copying the WebWorker EES, but I'm wondering why we decided to use an observable store here in the first place. We'll neither send it to the UI nor persist it, so it doesn't need to be serializable. Do we subscribe to it anywhere?

If it's indeed useless, we should remove it from both EES classes in a follow-up PR.

Reply:
The reason we had the observable store to begin with was to surface the `plugin -> worker` mapping in the UI in a task manager-like interface. We're going to decide what to do with it elsewhere.